### ISSUE TITLE
Clarify documentation of `editor.suggest.replaceMode`

### ISSUE THREAD

**Issue by hyangah** [microsoft/vscode#212390]
Currently `editor.suggest.replaceMode` setting description says:

"Controls whether words are overwritten when accepting completions.
Note that this depends on extensions opting into this feature."

Can you please provide what you mean by "extensions opting into this feature"?

Our extension users (vscode-go) reported `editor.suggest.replaceMode` is not respected,
but our extension does not override this setting at all.
I vaguely guessed this could be related to our language server not yet
returning `InsertReplaceEdit` (our LSP server sends `TextEdit` with snippet)
but I am not sure if that is the root cause.

**Comment by hyangah** [comment]
Looks like LSP `TextEdit` corresponds to vscode `CompletionItem` with old `Range`, and LSP `InsertReplaceEdit` corresponds to `CompletionItem` with the newer [`{ inserting, replacing}` range](https://code.visualstudio.com/api/references/vscode-api#CompletionItem.insertText).  
([LSP client code](https://github.com/microsoft/vscode-languageserver-node/blob/4b5f9cf622963dcfbc6129cdc1a570e2bb9f66a4/client/src/common/protocolConverter.ts#L644-L649))

Does the editor behave differently in `insert` and `replace` mode, if the extension always provides the old style completion items (that use `Range`)?

**Comment by jrieken** [comment]
> I vaguely guessed this could be related to our language server not yet returning InsertReplaceEdit (our LSP server sends TextEdit with snippet) but I am not sure if that is the root cause.

Yes, that's the root cause and that's the adoption needed on the extension side. These changes are usually announced in our release notes and I don't believe it is right to use the configuration description to hint extension authors to API they should use.

**Comment by hyangah** [comment]
Thanks! Yes, it makes sense not to use the config description for this, since the config description is for the end users.

How about mentioning the this `editor.suggest.insertMode` setting in the API documentation? For example somewhere in https://code.visualstudio.com/api/references/vscode-api#CompletionItem.

The `range` field description mentions the behavior when the field is omitted, but does not explain the difference in behavior when using the old `Range` or the `{inserting/replacing}` form. Release notes are great for advertising changes, but hard to search or use for newly joined developers.

**Comment by jrieken** [comment]
We have docs here https://github.com/microsoft/vscode/blob/f10bb2140999946ed845db28da29f65a75373ae7/src/vscode-dts/vscode.d.ts#L4967-L4987
